Mathlib/MeasureTheory/Integral/Layercake.lean

```diff
@@ -152,9 +150,20 @@ theorem lintegral_comp_eq_lintegral_meas_le_mul_of_measurable (μ : Measure α)
     · have h' : (p_fst, p_snd) ∉ {p : α × ℝ | p.snd ∈ Ioc 0 (f p.fst)} := h
       rw [Set.indicator_of_not_mem h', Set.indicator_of_not_mem h]
   rw [aux₂]
-  have mble := measurableSet_region_between_oc measurable_zero f_mble MeasurableSet.univ
-  simp_rw [mem_univ, Pi.zero_apply, true_and_iff] at mble
-  exact (ENNReal.measurable_ofReal.comp (g_mble.comp measurable_snd)).aemeasurable.indicator mble
+  have mble₀ : NullMeasurableSet {p : α × ℝ | p.snd ∈ Ioc 0 (f p.fst)} (μ.prod volume) := by
+    obtain ⟨f₀, ⟨mble_f₀, f_aeeq_f₀⟩⟩ := f_mble
+    have mble := measurableSet_region_between_oc measurable_zero mble_f₀ MeasurableSet.univ
+    apply NullMeasurableSet.congr mble.nullMeasurableSet
+    rw [eventuallyEq_iff_exists_mem] at *
+    obtain ⟨s, ⟨s_null, f_eq_f₀_on_s⟩⟩ := f_aeeq_f₀
+    refine ⟨s ×ˢ univ, ⟨?_, fun p hp ↦ by simp [setOf, f_eq_f₀_on_s hp.1]⟩⟩
+    simp only [Measure.ae, Filter.mem_mk, mem_setOf_eq] at *
+    suffices : (μ.prod (volume : Measure ℝ)) (sᶜ ×ˢ univ) = 0
+    · convert this
+      ext p
+      simp only [mem_compl_iff, mem_prod, mem_univ, and_true]
+    rw [Measure.prod_prod, s_null, zero_mul]
```

Comment:
Instead of having a block like that in the proof, could you state a lemma (extending `measurableSet_region_between_oc` to the ae measurable situation) that would be enough for your needs?

Reply:
Yes, that makes sense. I wasn't sure how often this would be needed, but splitting it off is certainly cleaner and possibly useful in the future. I guess the other variants `measurableSet_region_between_co`, `measurableSet_region_between_cc`, and `measurableSet_regionBetween` should be correspondingly generalized?

For now I did the basic variant as `nullMeasurableSet_regionBetween`, but I need another variant to actually refactor the proof. I will do that later this week.

Reply:
I now added a.e./null versions of `measurableSet_region_between_oc` and 3 friends in `MeasureTheory.Measure.Lebesgue.Basic` and the current longish proof can more or less just quote `nullMeasurableSet_region_between_oc` (a 2 lines `simpa only [...] using ...`).